### PR TITLE
Openneuro example and reporting utils

### PR DESCRIPTION
**Pull request by MartinPerez**
reporting utils include the images comparison plot, a contrast plot and a cluster stats table

### CODE REVIEW

File: nistats/reporting.py [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+import matplotlib 
+matplotlib.rc('xtick', labelsize=20) 
+
+
+def compare_niimgs(ref_imgs, src_imgs, masker, plot_hist=True, log=True,
+                   ref_label="image set 1", src_label="image set 2",
+                   output_dir=None, axes=None):
+    """Creates plots to compare two lists of images and measure correlation.
+
+    The first plot tests linear correlation between voxel values
```
Comment (bthirion) [inline nistats/reporting.py]: 'displays' rather thab 'tests'

File: nistats/reporting.py [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+    The second plot superimpose histograms to compare values distribution
```
Comment (bthirion) [inline nistats/reporting.py]: superimposes

File: nistats/reporting.py [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+
+    Parameters
+    ----------
+    ref_imgs: nifti_like
+        reference images.
+
+    src_imgs: nifti_like
+        Source images.
+
+    log: boolearn, optional (default True)
```
Comment (bthirion) [inline nistats/reporting.py]: Boolean

File: nistats/reporting.py [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+        Passed to plt.hist
+
+    plot_hist: boolean, optional (default True)
+        If True then then histograms of each img in ref_imgs will be plotted
```
Comment (bthirion) [inline nistats/reporting.py]: 'then' repeated

File: nistats/reporting.py [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+    ref_imgs: nifti_like
+        reference images.
+
+    src_imgs: nifti_like
+        Source images.
+
+    log: boolearn, optional (default True)
+        Passed to plt.hist
+
+    plot_hist: boolean, optional (default True)
```
Comment (bthirion) [inline nistats/reporting.py]: Boolean

File: examples/plot_bids_features.py [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+# Obtain automatically FirstLevelModel objects and fit arguments
+# ---------------------------------------------------------------
+# From the dataset directory we obtain automatically FirstLevelModel objects
+# with their subject_id filled from the BIDS dataset. Moreover we obtain
+# for each model a dictionary with run_imgs, events and confounder regressors
+# since in this case a confounds.tsv file is available in the BIDS dataset.
+# To get the first level models we have to specify the dataset directory,
+# the task_label and the space_label as specified in the file names.
+# We also have to provide the folder with the desired derivatives, that in this
+# were produced by the fmriprep BIDS app.
```
Comment (bthirion) [inline examples/plot_bids_features.py]: to be read again

File: nistats/thresholding.py [hunk 7]
```diff
@@ -108,14 +108,23 @@ def get_clusters_table(stat_img, stat_threshold, cluster_threshold):
 
     stat_map = stat_img.get_data()
 
+    # If the stat threshold is too high simply return an empty dataframe
+    if np.sum(stat_map > stat_threshold) == 0:
+        return pd.DataFrame()
```
Comment (bthirion) [inline nistats/thresholding.py]: Issue a warning here.

File: nistats/thresholding.py [hunk 8]
```diff
@@ -108,14 +108,23 @@ def get_clusters_table(stat_img, stat_threshold, cluster_threshold):
...
-    # labels = label_map[search_mask.get_data() > 0]
     for label_ in range(1, n_labels + 1):
         if np.sum(label_map == label_) < cluster_threshold:
             stat_map[label_map == label_] = 0
 
+    # If the cluster threshold is too high simply return an empty dataframe
+    # this checks for stats higher than threshold after small clusters
+    # were removed from stat_map
+    if np.sum(stat_map > stat_threshold) == 0:
+        return pd.DataFrame()
```
Comment (bthirion) [inline nistats/thresholding.py]: same thing.

File: examples/plot_bids_features.py (hunk 6 above)
Comment (MartinPerez) [reply examples/plot_bids_features.py]: this will be retouched in the release PR

**Comment by MartinPerez** [comment]
@bthirion all comments taken into account. all problems fixed, ready for merge.

File: examples/plot_bids_features.py (hunk 6 above)
Comment (bthirion) [reply examples/plot_bids_features.py]: 'in this case' ?

**Comment by bthirion** [comment]
LGTM overall. Merging

**Pull request merged**